File: 20-sortiranje/brzo-redjanje.py
def poredjaj(niz):
    if len(niz) <= 1: return niz

    manje = []
    jednako = []
    vece = []
    pivot = niz[0]
    for x in niz:
        if x < pivot:
            manje.append(x)
        if x == pivot:
            jednako.append(x)
        if x > pivot:
            vece.append(x)
    return poredjaj(manje) + jednako + poredjaj(vece)


print(poredjaj([54, 26, 93, 17, 77, 31, 44, 55, 20]))

File: 20-sortiranje/brzo-redjanje.js
const slucajniNiz = require('../utils/slucajniNiz')

function brzoRedjaj(niz) {
  if (niz.length <= 1) return niz

  const levo = []
  const desno = []
  const pivot = niz.pop()

  for (let i = 0; i < niz.length; i++) {
    if (niz[i] <= pivot) {
      levo.push(niz[i])
    } else {
      desno.push(niz[i])
    }
  }
  return [...brzoRedjaj(levo), pivot, ...brzoRedjaj(desno)]
}

console.log(brzoRedjaj(slucajniNiz(20)))
